oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/authentication/OAuth2LoginAuthenticationToken.java

```diff
@@ -45,10 +45,7 @@
 public class OAuth2LoginAuthenticationToken extends AbstractAuthenticationToken {
 	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
 	private OAuth2User principal;
-	private ClientRegistration clientRegistration;
-	private OAuth2AuthorizationExchange authorizationExchange;
-	private OAuth2AccessToken accessToken;
-	private OAuth2RefreshToken refreshToken;
+	private OAuth2AuthorizationCodeAuthenticationToken oAuth2AuthorizationCodeAuthenticationToken;
```

Comment:
Instead of composition use inheritance `OAuth2LoginAuthenticationToken extends OAuth2AuthorizationCodeAuthenticationToken` as noted in this [comment](https://github.com/spring-projects/spring-security/issues/5633#issuecomment-410059242)